aim2dat/strct/ext_manipulation/add_structure.py

```diff
@@ -0,0 +1,294 @@
+    host_indices : int or list (optional)
+        Index or indices of the host site(s). In case several indices are given the center of the
+        sites is defined as host site as reference point for the bond length.
+    guest_index : int (optional)
+        Index of the guest site.
+    guest_structure : str or aim2dat.strct.Structure (optional)
+        A representation of the guest structure given as a string of a functional group or molecule
+        (viable options are ``'CH3'``, ``'COOH'``, ``'H2O'``, ``'NH2'``, ``'NO2'`` or ``'OH'``), a
+        ``Structure`` object (bond direction is assumed to be the ``[-1.0, 0.0, 0.0]`` direction)
+        or the element symbol to add one single atom.
+    bond_length : float
+        Bond length between the host atom and the base atom of the functional group.
+    r_max : float (optional)
+        Cut-off value for the maximum distance between two atoms in angstrom.
+    cn_method : str (optional)
+        Method used to calculate the coordination environment.
+    min_dist_delta : float (optional)
+        Tolerance parameter that defines the relative distance from the nearest neighbour atom
+        for the ``'minimum_distance'`` method.
+    n_nearest_neighbours : int (optional)
+        Number of neighbours that are considered coordinated for the ``'n_neighbours'``
+        method.
+    econ_tolerance : float (optional)
+        Tolerance parameter for the econ method.
+    econ_conv_threshold : float (optional)
+        Convergence threshold for the econ method.
+    voronoi_weight_type : str (optional)
+        Weight type of the Voronoi facets. Supported options are ``'covalent_atomic_radius'``,
+        ``'area'`` and ``'solid_angle'``. The prefix ``'rel_'`` specifies that the relative
+        weights with respect to the maximum value of the polyhedron are calculated.
+    voronoi_weight_threshold : float (optional)
+        Weight threshold to consider a neighbouring atom coordinated.
+    dist_constraints : list (optional)
+        List of three-fold tuples containing the index of the site of the host structure, the index
+        of the site of the guest structure and the target distance. The position of the guest
+        structure is varied based on a grid search until the sum of the absolute errors in
+        minimized.
+    dist_threshold : float or None (optional)
+        Check the distances between all site pairs of the host and guest structure to ensure that
+        none of the added atoms collide.
+    change_label : bool (optional)
+        Add suffix to the label of the new structure highlighting the performed manipulation.
+
+    Returns
+    -------
+    aim2dat.strct.Structure
+        Structure with attached functional group.
+    """
+    guest_strct, guest_strct_label = _check_guest_structure(guest_structure)
+    if isinstance(host_indices, int):
+        host_indices = [host_indices]
+
+    if all(idx < len(structure) for idx in host_indices) and guest_index < len(
```

Comment:
I was wondering where the else statement is? This should give an error since the index is out of range, but this error will appear anyway if you try to run the function. Therefore, I would leave the if statement out

Reply:
This is expected behavior. The function should return None if the indices are out of range.

Reply:
Since this if statement is quite long, it might make sense to reverse the logic. Just handle the out of range case first (check it in an if statement) and return `None`. If this is not triggered, the function will anyway go ahead and return the new structure in the end. In this way, this large block is not indented and one doen't forget the purpose of the if statement